plugin/core/sessions.py

```diff
@@ -850,6 +851,9 @@ def m_workspace_configuration(self, params: Dict[str, Any], request_id: Any) ->
         for requested_item in requested_items:
             configuration = self.config.settings.copy(requested_item.get('section') or None)
             if self._plugin:
+                buffer_view = self.get_session_buffer_for_uri_async(requested_item['scopeUri'])
```

Comment:
If there's a `scopeUri`, then I think a more natural thing to do is to search for all of the views that match the scopeUri and then pass those views as a list to `on_workspace_configuration`. Since LSP-eslint is the only one using `on_workspace_configuration` can we just break API here?

Reply:
The languageId is also stored in the settings by the way:

```
>>> {k: v for k, v in view.settings().to_dict().items() if 'lsp' in k}
{'lsp_active': True, 'lsp_hover_provider_count': 2, 'lsp_language': {'clangd': 'cpp'}}
```

Reply:
That `lsp_language` entry is a dict for now, but it should be a plain string. https://github.com/sublimelsp/LSP/blob/31376ed0f5072144e9989dca8c406b14a57ffffb/plugin/session_view.py#L67-L74

(should be resolved once we have DocumentSelector in the config and lookups go through language-ids.sublime-settings)

Reply:
Let's not require plugins to fetch stuff from the view settings. That should stay internal to LSP so that we are free to change those in breaking ways. You said yourself that the languageId is gonna be in a dict structure eventually. The API should allow getting the language ID easily, not require digging through internal view settings.

As for passing the views, it might make sense from the conceptual point of view (since `scopeUri` theoretically be a URI of a directory also), but as far as getting the languageId it doesn't help much (because I don't want to use view settings).

Maybe we can pass a list of `SessionView` (or `SessionViewProtocol`) instead? Then it should be easy to expose languageId on it.

Reply:
I’m okay with passing SessionViewProtocol list :)

Reply:
Well, maybe in that case add another method to AbstractPlugin that gets all the SessionViewProtocols given an URI

Reply:
It makes more sense to return an Optional[SessionBufferProtocol] because you can access the list of SessionViewProtocols through that.

Reply:
Sounds good.

Reply:
Just realized that I have access to `Session` in `AbstractPlugin` instance and it has a `get_session_buffer_for_uri_async` method already. So I don't really need to add anything.